xbmc/URL.cpp

```diff
@@ -253,35 +253,29 @@ void CURL::Parse(const std::string& strURL1)
     }
   }
 
+  std::string strHostNameAndPort = strURL.substr(iPos, (iSlash == std::string::npos) ? iEnd - iPos : iSlash - iPos);
+  // check for IPv6 numerical representation inside []
+  size_t iBrk = strHostNameAndPort.rfind("]");
+  if (iBrk < iEnd && iBrk != std::string::npos && strHostNameAndPort.find("[") == 0)
+  {
+    m_strHostName = strHostNameAndPort.substr(1, iBrk-1);
+    strHostNameAndPort.erase(0, iBrk+1);
+  }
+
   // detect hostname:port/
-  if (iSlash == std::string::npos)
+  size_t iColon = strHostNameAndPort.rfind(":");
+  if (iColon != std::string::npos && iColon == strHostNameAndPort.find(":"))
```

Comment:
Why do you think?
At this point (trying to to find port definition) iBrk has no meaning anymore. As it either was 'npos' (no ']') or the pair '[]' was found, and hostname/ip was stripped out (while removing '[]' too).

The position of iBrk is now literally -1 of the new, shortened strHostNameAndPort.